### PR TITLE
GEODE-2365: Replace gemfire with geode in clicache src.

### PR DESCRIPTION
**Pull request by mmartell**
Note: There are some reformats being done by Visual Studio in both the csharp and the c++/cli sources.

### CODE REVIEW

File: src/clicache/integration-test/CacheHelperN.cs [hunk 1]
```diff
@@ -1877,7 +1878,7 @@ public static int getBaseLocatorPort()
           }
           try
           {
-            TextWriter tw = new StreamWriter(Directory.GetCurrentDirectory() + "\\" + startDir + "\\gemfire.properties", false);
+            TextWriter tw = new StreamWriter(Directory.GetCurrentDirectory() + "\\" + startDir + "\\geode.properties", false);
```
Comment (echobravopapa) [inline src/clicache/integration-test/CacheHelperN.cs]: Does this impact integration tests?

Comment (mikemartell) [inline src/clicache/integration-test/CacheHelperN.cs]: This is an output properties file created by the tests and is OK to call geode.properties. Thanks for checking though.

**Comment by mmartell** [comment]
Fixed one failing clicache (ThinClientCSTXN) due to an invalid gfsh arg. Apparently gfsh still requires gemfire terminology:
   gemfire.ALLOW_PERSISTENT_TRANSACTIONS
   not geode.ALLOW_PERSISTENT_TRANSACTIONS

**Comment by pivotal-jbarrett** [comment]
Merged, please close

**Comment by mmartell** [comment]
This closes #375.

**Pull request closed without merging**